src/Chapter4.hs

```diff
@@ -307,6 +326,10 @@ data List a
     = Empty
     | Cons a (List a)
 
+instance Functor List where
+  fmap :: (a -> b) -> List a -> List b
+  fmap _ Empty = Empty
+  fmap f (Cons head tail) = Cons (f head) (fmap f tail)
```

Comment:
`head` and `tail` are the names of standard functions. Haskell allows name shadowing, that's why it works. Though in practice, it's usually better to avoid name clashes with other functions cause it could be confusing 🙂 For lists, typical names are `x` for head and `xs` for tail.